### ISSUE TITLE
Missing HRV-channel StackedAreaDefinition for native_msg-reader

### ISSUE THREAD

**Issue by sjoro** [pytroll/satpy#299]
#### Code Sample, a minimal, complete, and verifiable piece of code

```python
f = MSG4-SEVI-MSG15-0100-NA-20180502124243.556000000Z-NA.nat
nat = Scene(f, reader='native_msg')
nat.load(['HRV'])
tmp = nat.resample("seviri_0deg")
tmp.show(param)
```
#### Problem description
Resampling native format SEVIRI HRV-channel onto `seviri_0deg`-area produces a strange looking result. HRV channel requires a special `StackedAreaDefinition` as it is scanned in two parts.

#### Expected Output
Correctly resampled HRV channel ;)

**Comment by marfel** [comment]
I had a similar issue, except my target area is Central Europe only. To be able to properly remap the HRV channel, I transformed the HRV array into `<xarray.DataArray (y: 11136, x: 11136)>`, using the header information to properly fill in the scanlines. Then, it can be mapped exactly like the other channels. 
Sorry this is a bit of a hack, and I haven't got the time to work through your contributer's guide right now, but here is a little patch that does this for me. Maybe it can be of use to someone.
[satpy_patch.txt](https://github.com/pytroll/satpy/files/2093099/satpy_patch.txt)

**Comment by sjoro** [comment]
@mraspaud yes, i'll have a look later this week or early next week. can't promise that I can get it working before the end of next week.

**Comment by Nitrofest** [comment]
Hi all,
I am not sure if my issue is the same. Could someone help, please? I am using Satpy to read the native format ".msg" files. I prefer to make the plots by myself, so I retrieve both reflectances and lat/lon data, and plot the RGB composite passing to Matplotlib's pcolormesh. Up to now, I've been working with full disk data and there was no problem. I read the lat/lon data with:

latlon=scn['VIS008'].area.get_lonlats()
lat=np.asarray(latlon[0])
lon=np.asarray(latlon[1])

Now, I need to read the HRV channel, and it seems that it has no lat/lon coordinates within the file:

latlon=scn['HRV'].area.get_lonlats()
AttributeError: 'DataArray' object has no attribute 'area' 

scn['HRV']
<xarray.DataArray (y: 11136, x: 5568)>
Dimensions without coordinates: y, x
Attributes:
    units:                %
    wavelength:           (0.5, 0.7, 0.9)
    sensor:               seviri
    platform_name:        Meteosat-10
    standard_name:        toa_bidirectional_reflectance
    start_time:           [datetime.datetime(2017, 7, 24, 14, 45, 9, 772925)]
    end_time:             [datetime.datetime(2017, 7, 24, 15, 0, 9, 451257)]
    name:                 HRV
    resolution:           1000.134348869
    calibration:          reflectance
    polarization:         None
    level:                None
    modifiers:            ()
    ancillary_variables:  []

Could someone advise please how can I get the lat/lon coordinates for HRV channel?

**Comment by marfel** [comment]
You could probably use the little patch I supplied a few comments back (I think it has not found its way into the code yet). Then for HRV you get an xarray of 11136 x 11136, for which you can calculate the coordinates by interpolating the VIS coordinates to the 11136 grid. No idea how accurate this, but for colorful pictures it should be ok, I guess ;)

**Comment by mraspaud** [comment]
It's still of our todo list, so it will be implemented eventually. I think I will have a look at this rather sooner than later (within a couple of weeks), but @marfel 's patch can be used in the meanwhile of course.

**Comment by Nitrofest** [comment]
Thanks everyone, I was able to get the coordinates after using this patch
https://github.com/pytroll/satpy/pull/376

**Comment by djhoese** [comment]
@martin @sjoro any updates on this?

**Comment by sjoro** [comment]
@djhoese sorry, not from my side.

**Comment by mraspaud** [comment]
Will we have something to release on monday ? ;)

**Comment by sjoro** [comment]
ehrm... no. i've earmarked this enhancement for my todo list for Madison.

**Comment by mraspaud** [comment]
ok, good to know :) thanks

**Comment by ameraner** [comment]
Hi all, I think this issue was closed by #985, and further improved by #1196.

**Comment by mraspaud** [comment]
Perfect, closing this then.
